### DISCUSSION TITLE
kpimoni xapp not working as intended

### DISCUSSION THREAD

**Discussion by joshuamoorexyz** [srsran/srsRAN_Project#205] [General Help]
## Issue Description ##
When running the instructions here https://docs.srsran.com/projects/project/en/latest/tutorials/source/flexric/source/index.html# for a zmq setup there is an issue when running the kpimoni xapp. 

The xapp prints: [E2AP]: Resending Setup Request after timeout

and the nearRT RIC crashes with the message: 

joshuamoorexyz@joshuamoorexyz-B550-UD-AC:~/Github/flexric/build/examples/ric$ sudo ./nearRT-RIC 
Setting the config -c file to /usr/local/etc/flexric/flexric.conf
Setting path -p for the shared libraries to /usr/local/lib/flexric/
[NEAR-RIC]: nearRT-RIC IP Address = 127.0.0.1, PORT = 36421
[NEAR-RIC]: Initializing 
[NEAR-RIC]: Loading SM ID = 143 with def = RLC_STATS_V0 
[NEAR-RIC]: Loading SM ID = 144 with def = PDCP_STATS_V0 
[NEAR-RIC]: Loading SM ID = 145 with def = SLICE_STATS_V0 
[NEAR-RIC]: Loading SM ID = 146 with def = TC_STATS_V0 
[NEAR-RIC]: Loading SM ID = 147 with def = ORAN-E2SM-KPM 
[NEAR-RIC]: Loading SM ID = 142 with def = MAC_STATS_V0 
[NEAR-RIC]: Loading SM ID = 148 with def = GTP_STATS_V0 
[iApp]: Initializing ... 
[iApp]: nearRT-RIC IP Address = 127.0.0.1, PORT = 36422
fd created with 6 
[iApp]: E42 SETUP-REQUEST received
[iApp]: E42 SETUP-RESPONSE sent
nearRT-RIC: /home/joshuamoorexyz/Github/flexric/src/lib/ap/enc/e2ap_msg_enc_asn.c:2670: e2ap_enc_e42_setup_response_asn_pdu: Assertion `sr->len_e2_nodes_conn > 0 && "No global node conected??"' failed.
Aborted

## Setup Details ##
[Specify details of the test setup. This would help us reproduce the problem reliably]
e.g. Network configuration, Operation System, Hardware, RF front-end, library and driver versions

Ubuntu 22.04
Following all instructions for configuration of open5gs, gnb and 4g srsue. This setup works other than the xapp.

## Expected Behavior ##
[What you expect to happen]

The xapp to recieve the E2 Setup response and perform its monitoring functions.

**Comment by pgawlowicz** [comment]
Hi, did you manage to solve it? did you apply the patch from our docs?

**Reply by joshuamoorexyz** [reply]
I did not solve it and yes i did apply the patch.Not sure what the error no global node connected is referring either.

**Reply by alektoja** [reply]
@pgawlowicz is the patch already generally available in the latest version? I encountered a similar problem with the latest commit.

**Reply by pgawlowicz** [reply]
the patch is not needed with `br-flexric` branch and commit: `1a3903a7`.

**Comment by yagoda** [comment]
Can you search the logs for "E2" and post the output please?

**Reply by joshuamoorexyz** [reply]
i tried it on a new PC actually and when i start the GnB the nearRT RIC crashes with this error 


joshuamoorexyz@joshuamoorexyz-B550-UD-AC:~/Github/flexric/build/examples/ric$ sudo ./nearRT-RIC 
Setting the config -c file to /usr/local/etc/flexric/flexric.conf
Setting path -p for the shared libraries to /usr/local/lib/flexric/
[NEAR-RIC]: nearRT-RIC IP Address = 127.0.0.1, PORT = 36421
[NEAR-RIC]: Initializing 
[NEAR-RIC]: Loading SM ID = 143 with def = RLC_STATS_V0 
[NEAR-RIC]: Loading SM ID = 144 with def = PDCP_STATS_V0 
[NEAR-RIC]: Loading SM ID = 145 with def = SLICE_STATS_V0 
[NEAR-RIC]: Loading SM ID = 146 with def = TC_STATS_V0 
[NEAR-RIC]: Loading SM ID = 147 with def = ORAN-E2SM-KPM 
[NEAR-RIC]: Loading SM ID = 142 with def = MAC_STATS_V0 
[NEAR-RIC]: Loading SM ID = 148 with def = GTP_STATS_V0 
[iApp]: Initializing ... 
[iApp]: nearRT-RIC IP Address = 127.0.0.1, PORT = 36422
fd created with 6 
Received message with id = 411, port = 27337 
[E2AP] Received SETUP-REQUEST from PLMN   1. 1 Node ID 411 RAN type ngran_gNB
nearRT-RIC: /home/joshuamoorexyz/Github/flexric/src/ric/msg_handler_ric.c:88: generate_setup_response: Assertion `req->ran_func_item[i].def.len < 127' failed.
Aborted

**Reply by pgawlowicz** [reply]
You need to apply the FlexrRIC patch that available in the app note. See [here](https://docs.srsran.com/projects/project/en/latest/tutorials/source/flexric/source/index.html#id2)

**Comment by yagoda** [comment]
This is a size assertion error, the patch should fix this, go to the line
in the code (flexric/src/ric/msg_handler_ric.c:88:) and increase the size
in the assert, be sure to increase the size of whatever container it
depends on too.

**Comment by joshuamoorexyz** [comment]
This did fix it and i was able to run KPI moni xapp! The original post that i had here i was trying with open5gs running on the machine from the daemon. Is it currently not supported to do this since i had the error like before? specifically nearRT-RIC: /home/joshuamoorexyz/Github/flexric/src/lib/ap/enc/e2ap_msg_enc_asn.c:2670: e2ap_enc_e42_setup_response_asn_pdu: Assertion `sr->len_e2_nodes_conn > 0 && "No global node conected??"' failed.
Aborted i can try to recreate if it would need more logs to debug. Thanks

**Reply by pgawlowicz** [reply]
Please check the updated FlexRIC - srsRAN app note.
